### PR TITLE
Here are the new care plan

### PR DESCRIPTION
**Pull request by juggernautsei**
Signed-off-by: Sherwin Gaddis <sherwingaddis@gmail.com>

Here is another commit that has been mentored by ... Tony!!!!

### CODE REVIEW

**Comment by tmccormi** [comment]
THis is much better.   You seem to have copied an older version of globals.php over the current version, however.   Please get the latest version from the master and merge your module changes with that and then update the branch

**Comment by juggernautsei** [comment]
Ayh ayh captain!!
Updated

File: library/globals.inc.php [hunk 1]
```diff
@@ -2211,13 +2217,13 @@
             'ms' => xl('Malay'),
             'ml' => xl('Malayalam'),
             'mt' => xl('Maltese'),
-            'mi' => xl('Māori'),
-            'mr' => xl('Marathi (Marāṭhī)'),
```
Comment (bradymiller) [inline library/globals.inc.php]: Watch out here. Ensure your text editor is working in UTF8 mode or else you'll break characters as is done here.

File: sql/4_1_2-to-4_2_0_upgrade.sql [hunk 2]
```diff
@@ -131,6 +131,65 @@ CREATE TABLE `erx_ttl_touch` (
...
+  `prevention` text NOT NULL,
+  `pmh` text NOT NULL,
+  `psh` varchar(255) NOT NULL,
+  `fhsh` text NOT NULL,
+  `sh` varchar(255) NOT NULL,
+  `audit` int(1) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id` (`id`),
+  KEY `pid` (`pid`)
+) ENGINE=InnoDB  DEFAULT CHARSET=utf8 AUTO_INCREMENT=66 ;
```
Comment (bradymiller) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: Do not declare the DEFAULT CHARSET.

File: sql/4_1_2-to-4_2_0_upgrade.sql [hunk 3]
```diff
@@ -131,6 +131,65 @@ CREATE TABLE `erx_ttl_touch` (
...
+
+#IfNotTable care_plan_audit
+CREATE TABLE IF NOT EXISTS `care_plan_audit` (
+  `id` int(5) NOT NULL AUTO_INCREMENT,
+  `user` varchar(10) NOT NULL,
+  `date` date NOT NULL,
+  `pid` varchar(5) NOT NULL,
+  `encounter` varchar(5) NOT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB  DEFAULT CHARSET=utf8 AUTO_INCREMENT=6 ;
```
Comment (bradymiller) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: Do not declare the DEFAULT CHARSET.

File: interface/main/main_title.php [hunk 4]
```diff
@@ -87,6 +87,12 @@ function showhideMenu() {
             <a href='' class="css_button_small" style="margin:0px;vertical-align:top;display:none;" id='clear_active' onClick="javascript:parent.left_nav.clearactive();return false;">
```
Comment (bradymiller) [inline interface/main/main_title.php]: What is the above database.sql file doing. Should be modifying the file sql/database.sql instead.

File: library/js/fancybox.2.1.5/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
+fancyBox - Changelog
```
Comment (bradymiller) [inline library/js/fancybox.2.1.5/CHANGELOG.md]: What is the license that fancybox uses?

Comment (bradymiller) [inline library/js/fancybox.2.1.5/CHANGELOG.md]: License doesn't look promising:
http://fancyapps.com/fancybox/#license

File: sql/4_1_2-to-4_2_0_upgrade.sql [hunk 6]
```diff
@@ -2856,6 +3020,36 @@ UPDATE `clinical_rules` SET `amc_2014_flag` = 1 , `amc_code_2014` = '170.314(g)(
...
+#IfNotRow2D layout_options form_id LBFPHEX field_id breast
+INSERT INTO `layout_options` (`form_id`, `field_id`, `group_name`, `title`, `seq`, `data_type`, `uor`, `fld_length`, `max_length`, `list_id`, `titlecols`, `datacols`, `default_value`, `edit_options`, `description`, `fld_rows`, `list_backup_id`, `source`, `conditions`) VALUES
+('LBFPHEX', 'breast', '1Normal', 'Breasts', 60, 25, 1, 55, 0, 'PHEXGEN6', 10, 10, '', '', '', 0, '', 'F', ''),
+('LBFPHEX', 'gen', '1Normal', 'General', 10, 25, 1, 55, 0, 'PHEXGEN', 10, 10, '', '', '', 0, '', 'F', ''),
+('LBFPHEX', 'heart', '1Normal', 'Heart', 40, 25, 1, 55, 0, 'PHEXGEN4', 10, 10, '', '', '', 0, '', 'F', ''),
+('LBFPHEX', 'heent', '1Normal', 'HEENT', 20, 25, 1, 55, 0, 'PHEXGEN2', 10, 10, '', '', '', 0, '', 'F', ''),
+('LBFPHEX', 'lungs', '1Normal', 'Lungs', 50, 25, 1, 55, 0, 'PHEXGEN5', 10, 10, '', '', '', 0, '', 'F', ''),
+('LBFPHEX', 'neck', '1Normal', 'Neck/Nodes', 30, 25, 1, 55, 0, 'PHEXGEN3', 10, 10, '', '', '', 0, '', 'F', '');
+#EndIf
+
```
Comment (bradymiller) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: It appears you may be not using the standard way to store labs in OpenEMR? Is that true.

File: sql/4_1_2-to-4_2_0_upgrade.sql [hunk 7]
```diff
@@ -2856,6 +3020,36 @@ UPDATE `clinical_rules` SET `amc_2014_flag` = 1 , `amc_code_2014` = '170.314(g)(
...
 #IfNotRow2D layout_options form_id HIS field_id dc_father
```
Comment (bradymiller) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: My initial instinct is that using LBF forms seems a bit awkard, but will see what other developers think.

File: interface/patient_file/summary/demographics.php [hunk 8]
```diff
@@ -1381,24 +1375,19 @@ function setMyPatient() {
...
         }
-        if ($resNotNull) { //////
+        if (isset($res) && $res != null) {
             if ( $count < 1 ) { 
                 echo "&nbsp;&nbsp;" . htmlspecialchars(xl('None'),ENT_NOQUOTES); 
             }
             echo "</div>";
-        }
       }
+    }
```
Comment (bradymiller) [inline interface/patient_file/summary/demographics.php]: Watch out. Others code is getting reverted here.

File: interface/patient_file/summary/rx_noframeset.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+      height:80%;
+    }
+  </style>
+</head>
+<body>
+
+ <iframe class="menu" src="rx_left.php" name="RxLeft" scrolling="auto"></iframe>
+ <iframe class="mainContent" src="<?php echo $GLOBALS['webroot'] ?>/controller.php?prescription&list&id=<?php echo $pid ?>"
+  name="RxRight" scrolling="auto"></iframe>
+
```
Comment (bradymiller) [inline interface/patient_file/summary/rx_noframeset.php]: Is this related to the care plan?

File: sql/4_1_2-to-4_2_0_upgrade.sql (hunk 2 above)
Comment (bradymiller) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: Rather than hard-code a maximum of 5 for each of the items, wouldn't a database structure that allows as many as want be better. For example, a care_plan_element table row would contain a diag, active, risk, etc, and a mapping id back to the id of the care_plan table element, which houses the main things like id, date, encounter.

File: sql/4_1_2-to-4_2_0_upgrade.sql [hunk 10]
```diff
@@ -2856,6 +3020,36 @@ UPDATE `clinical_rules` SET `amc_2014_flag` = 1 , `amc_code_2014` = '170.314(g)(
...
 	INSERT INTO `layout_options` (`form_id`, `field_id`, `group_name`, `title`, `seq`, `data_type`, `uor`, `fld_length`, `max_length`, `list_id`, `titlecols`, `datacols`, `default_value`, `edit_options`, `description`, `fld_rows`, `list_backup_id`) VALUES ('HIS', 'dc_father', '2Family History', 'Diagnosis Code', 2, 15, 1, 0, 255, '', 1, 1, '', '', '', 0, '');
         UPDATE `layout_options` SET `seq` = '3' WHERE `layout_options`.`form_id` = 'HIS' AND `layout_options`.`field_id` = 'history_mother';
```
Comment (bradymiller) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: Did a quick review with some comment/thoughts.
Known big issue:
1. Is fancybox license compatible with GPL?
Known small issue:
1. Will need to incorporate translations, html escaping, and sql binding

File: interface/main/main_title.php (hunk 4 above)
Comment (juggernautsei) [inline interface/main/main_title.php]: I have moved it to the sql folder.

File: library/js/fancybox.2.1.5/CHANGELOG.md (hunk 5 above)
Comment (juggernautsei) [inline library/js/fancybox.2.1.5/CHANGELOG.md]: I have place a request to fancy box to find the answer to this question

File: sql/4_1_2-to-4_2_0_upgrade.sql (hunk 2 above)
Comment (juggernautsei) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: Those were my thought for version two because this one was built on the fly and not much for thought given to database and best design practices. This would have been better built with OOP than hard coding but the project was on a very tight time table and was built in a hurry. So, hind sight now is where we are all working and given more time this can be built into a very versatile tool. These are just a humble beginnings. 

There are other things that could be done smarter given more time to develop.

File: sql/4_1_2-to-4_2_0_upgrade.sql (hunk 6 above)
Comment (juggernautsei) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: This is true.
This particular doctor had a specific request for labs and was built to meet those requirements. Did not have time to dig in and find out how labs are constructed.

File: sql/4_1_2-to-4_2_0_upgrade.sql (hunk 7 above)
Comment (juggernautsei) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: I understand.

File: sql/4_1_2-to-4_2_0_upgrade.sql (hunk 10 above)
Comment (juggernautsei) [inline sql/4_1_2-to-4_2_0_upgrade.sql]: I'll work on these things.

**Comment by Wakie87** [comment]
Hi @juggernautsei,

We thank you for your submitting a pull request to OpenEmr. Without contributions like this from the community these types of projects go stale.

Currently we are going through and doing a spring clean of old and outdated pull requests. We are giving contributors the chance to update the code they submitted if they think it is still relevant, close the pull requests themselves, or if we do not hear back within 3 days we will close the pull request ourselves. If you believe that closing the pull request is an error, let us know and we will look at re-opening it.

In case you do not already know, there is some documentation on the wiki for developers here: http://open-emr.org/wiki/index.php/OpenEMR_Wiki_Home_Page#Developer_Manuals

Just remember every single bit you contribute to an open source project counts. The more you contribute the better, but even fixing one small typo in a translation will make the final product better.

Thanks

The OpenEMR Team
http://www.open-emr.org/

**Comment by juggernautsei** [comment]
Since the care plan from ZH has been adopted, that makes this care plan obsolete in my eyes. I wish not to pursue this code. It can be dropped.
